src/vs/workbench/services/dialogs/browser/abstractFileDialogService.ts

```diff
@@ -77,6 +82,10 @@ export abstract class AbstractFileDialogService implements IFileDialogService {
 		}
 
 		if (!candidate) {
+			const shouldUseCwd = await this.configurationService.getValue<boolean>('window.cwdDefaultPath');
+			if (shouldUseCwd && (env['VSCODE_CWD'] || cwd())) {
+				return this.pathService.fileURI(env['VSCODE_CWD'] || cwd());
+			}
```

Comment:
This looks to be a direct duplicate of the the code above. It can be refactored into a separate method.